// notification-handler/bundles/notification-handler.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@angular/material/snack-bar'), require('@angular/common'), require('@angular/material/button'), require('@angular/material/icon'), require('@angular/material//snack-bar')) :
    typeof define === 'function' && define.amd ? define('notification-handler', ['exports', '@angular/core', '@angular/material/snack-bar', '@angular/common', '@angular/material/button', '@angular/material/icon', '@angular/material//snack-bar'], factory) :
    (global = global || self, factory(global['notification-handler'] = {}, global.ng.core, global.ng.material.snackBar, global.ng.common, global.ng.material.button, global.ng.material.icon, global.ng.material[''].snackBar));
}(this, (function (exports, i0, i1, i2, i3, i4, i1$1) { 'use strict';

    // TODO: Log error to the server like Rollbar or Crashlytics
    // TODO: treat the stack when error occurs
    // TODO: set the application name
    var LoggingService = /** @class */ (function () {
        function LoggingService() {
        }
        LoggingService.prototype.logError = function (message) {
            console.error(this.formatMessage(message));
        };
        LoggingService.prototype.logWarning = function (message) {
            console.warn(this.formatMessage(message));
        };
        LoggingService.prototype.logInformation = function (message) {
            console.log("Info: " + this.formatMessage(message));
        };
        LoggingService.prototype.logIt = function (message) {
            console.log(this.formatMessage(message));
        };
        LoggingService.prototype.formatMessage = function (message) {
            var now = new Date();
            return now.toLocaleString() + ": LoggingService: " + message;
        };
        return LoggingService;
    }());
    LoggingService.ɵfac = function LoggingService_Factory(t) { return new (t || LoggingService)(); };
    LoggingService.ɵprov = i0.ɵɵdefineInjectable({ token: LoggingService, factory: LoggingService.ɵfac });
    /*@__PURE__*/ (function () {
        i0.ɵsetClassMetadata(LoggingService, [{
                type: i0.Injectable
            }], null, null);
    })();

    /*! *****************************************************************************
    Copyright (c) Microsoft Corporation.

    Permission to use, copy, modify, and/or distribute this software for any
    purpose with or without fee is hereby granted.

    THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH
    REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
    AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
    INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
    LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR
    OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
    PERFORMANCE OF THIS SOFTWARE.
    ***************************************************************************** */
    /* global Reflect, Promise */
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b)
                if (b.hasOwnProperty(p))
                    d[p] = b[p]; };
        return extendStatics(d, b);
    };
    function __extends(d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    }
    var __assign = function () {
        __assign = Object.assign || function __assign(t) {
            for (var s, i = 1, n = arguments.length; i < n; i++) {
                s = arguments[i];
                for (var p in s)
                    if (Object.prototype.hasOwnProperty.call(s, p))
                        t[p] = s[p];
            }
            return t;
        };
        return __assign.apply(this, arguments);
    };
    function __rest(s, e) {
        var t = {};
        for (var p in s)
            if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
                t[p] = s[p];
        if (s != null && typeof Object.getOwnPropertySymbols === "function")
            for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
                if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                    t[p[i]] = s[p[i]];
            }
        return t;
    }
    function __decorate(decorators, target, key, desc) {
        var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
            r = Reflect.decorate(decorators, target, key, desc);
        else
            for (var i = decorators.length - 1; i >= 0; i--)
                if (d = decorators[i])
                    r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    }
    function __param(paramIndex, decorator) {
        return function (target, key) { decorator(target, key, paramIndex); };
    }
    function __metadata(metadataKey, metadataValue) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
            return Reflect.metadata(metadataKey, metadataValue);
    }
    function __awaiter(thisArg, _arguments, P, generator) {
        function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
        return new (P || (P = Promise))(function (resolve, reject) {
            function fulfilled(value) { try {
                step(generator.next(value));
            }
            catch (e) {
                reject(e);
            } }
            function rejected(value) { try {
                step(generator["throw"](value));
            }
            catch (e) {
                reject(e);
            } }
            function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
            step((generator = generator.apply(thisArg, _arguments || [])).next());
        });
    }
    function __generator(thisArg, body) {
        var _ = { label: 0, sent: function () { if (t[0] & 1)
                throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
        return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function () { return this; }), g;
        function verb(n) { return function (v) { return step([n, v]); }; }
        function step(op) {
            if (f)
                throw new TypeError("Generator is already executing.");
            while (_)
                try {
                    if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done)
                        return t;
                    if (y = 0, t)
                        op = [op[0] & 2, t.value];
                    switch (op[0]) {
                        case 0:
                        case 1:
                            t = op;
                            break;
                        case 4:
                            _.label++;
                            return { value: op[1], done: false };
                        case 5:
                            _.label++;
                            y = op[1];
                            op = [0];
                            continue;
                        case 7:
                            op = _.ops.pop();
                            _.trys.pop();
                            continue;
                        default:
                            if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) {
                                _ = 0;
                                continue;
                            }
                            if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) {
                                _.label = op[1];
                                break;
                            }
                            if (op[0] === 6 && _.label < t[1]) {
                                _.label = t[1];
                                t = op;
                                break;
                            }
                            if (t && _.label < t[2]) {
                                _.label = t[2];
                                _.ops.push(op);
                                break;
                            }
                            if (t[2])
                                _.ops.pop();
                            _.trys.pop();
                            continue;
                    }
                    op = body.call(thisArg, _);
                }
                catch (e) {
                    op = [6, e];
                    y = 0;
                }
                finally {
                    f = t = 0;
                }
            if (op[0] & 5)
                throw op[1];
            return { value: op[0] ? op[1] : void 0, done: true };
        }
    }
    var __createBinding = Object.create ? (function (o, m, k, k2) {
        if (k2 === undefined)
            k2 = k;
        Object.defineProperty(o, k2, { enumerable: true, get: function () { return m[k]; } });
    }) : (function (o, m, k, k2) {
        if (k2 === undefined)
            k2 = k;
        o[k2] = m[k];
    });
    function __exportStar(m, exports) {
        for (var p in m)
            if (p !== "default" && !exports.hasOwnProperty(p))
                __createBinding(exports, m, p);
    }
    function __values(o) {
        var s = typeof Symbol === "function" && Symbol.iterator, m = s && o[s], i = 0;
        if (m)
            return m.call(o);
        if (o && typeof o.length === "number")
            return {
                next: function () {
                    if (o && i >= o.length)
                        o = void 0;
                    return { value: o && o[i++], done: !o };
                }
            };
        throw new TypeError(s ? "Object is not iterable." : "Symbol.iterator is not defined.");
    }
    function __read(o, n) {
        var m = typeof Symbol === "function" && o[Symbol.iterator];
        if (!m)
            return o;
        var i = m.call(o), r, ar = [], e;
        try {
            while ((n === void 0 || n-- > 0) && !(r = i.next()).done)
                ar.push(r.value);
        }
        catch (error) {
            e = { error: error };
        }
        finally {
            try {
                if (r && !r.done && (m = i["return"]))
                    m.call(i);
            }
            finally {
                if (e)
                    throw e.error;
            }
        }
        return ar;
    }
    function __spread() {
        for (var ar = [], i = 0; i < arguments.length; i++)
            ar = ar.concat(__read(arguments[i]));
        return ar;
    }
    function __spreadArrays() {
        for (var s = 0, i = 0, il = arguments.length; i < il; i++)
            s += arguments[i].length;
        for (var r = Array(s), k = 0, i = 0; i < il; i++)
            for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
                r[k] = a[j];
        return r;
    }
    ;
    function __await(v) {
        return this instanceof __await ? (this.v = v, this) : new __await(v);
    }
    function __asyncGenerator(thisArg, _arguments, generator) {
        if (!Symbol.asyncIterator)
            throw new TypeError("Symbol.asyncIterator is not defined.");
        var g = generator.apply(thisArg, _arguments || []), i, q = [];
        return i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i;
        function verb(n) { if (g[n])
            i[n] = function (v) { return new Promise(function (a, b) { q.push([n, v, a, b]) > 1 || resume(n, v); }); }; }
        function resume(n, v) { try {
            step(g[n](v));
        }
        catch (e) {
            settle(q[0][3], e);
        } }
        function step(r) { r.value instanceof __await ? Promise.resolve(r.value.v).then(fulfill, reject) : settle(q[0][2], r); }
        function fulfill(value) { resume("next", value); }
        function reject(value) { resume("throw", value); }
        function settle(f, v) { if (f(v), q.shift(), q.length)
            resume(q[0][0], q[0][1]); }
    }
    function __asyncDelegator(o) {
        var i, p;
        return i = {}, verb("next"), verb("throw", function (e) { throw e; }), verb("return"), i[Symbol.iterator] = function () { return this; }, i;
        function verb(n, f) { i[n] = o[n] ? function (v) { return (p = !p) ? { value: __await(o[n](v)), done: n === "return" } : f ? f(v) : v; } : f; }
    }
    function __asyncValues(o) {
        if (!Symbol.asyncIterator)
            throw new TypeError("Symbol.asyncIterator is not defined.");
        var m = o[Symbol.asyncIterator], i;
        return m ? m.call(o) : (o = typeof __values === "function" ? __values(o) : o[Symbol.iterator](), i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i);
        function verb(n) { i[n] = o[n] && function (v) { return new Promise(function (resolve, reject) { v = o[n](v), settle(resolve, reject, v.done, v.value); }); }; }
        function settle(resolve, reject, d, v) { Promise.resolve(v).then(function (v) { resolve({ value: v, done: d }); }, reject); }
    }
    function __makeTemplateObject(cooked, raw) {
        if (Object.defineProperty) {
            Object.defineProperty(cooked, "raw", { value: raw });
        }
        else {
            cooked.raw = raw;
        }
        return cooked;
    }
    ;
    var __setModuleDefault = Object.create ? (function (o, v) {
        Object.defineProperty(o, "default", { enumerable: true, value: v });
    }) : function (o, v) {
        o["default"] = v;
    };
    function __importStar(mod) {
        if (mod && mod.__esModule)
            return mod;
        var result = {};
        if (mod != null)
            for (var k in mod)
                if (Object.hasOwnProperty.call(mod, k))
                    __createBinding(result, mod, k);
        __setModuleDefault(result, mod);
        return result;
    }
    function __importDefault(mod) {
        return (mod && mod.__esModule) ? mod : { default: mod };
    }
    function __classPrivateFieldGet(receiver, privateMap) {
        if (!privateMap.has(receiver)) {
            throw new TypeError("attempted to get private field on non-instance");
        }
        return privateMap.get(receiver);
    }
    function __classPrivateFieldSet(receiver, privateMap, value) {
        if (!privateMap.has(receiver)) {
            throw new TypeError("attempted to set private field on non-instance");
        }
        privateMap.set(receiver, value);
        return value;
    }

    function NotificationHandlerComponent_div_3_Template(rf, ctx) {
        if (rf & 1) {
            var _r2_1 = i0.ɵɵgetCurrentView();
            i0.ɵɵelementStart(0, "div", 3);
            i0.ɵɵelementStart(1, "button", 4);
            i0.ɵɵlistener("click", function NotificationHandlerComponent_div_3_Template_button_click_1_listener() { i0.ɵɵrestoreView(_r2_1); var ctx_r1 = i0.ɵɵnextContext(); return ctx_r1.snackBarRef.dismiss(); });
            i0.ɵɵelementStart(2, "mat-icon");
            i0.ɵɵtext(3, "close");
            i0.ɵɵelementEnd();
            i0.ɵɵelementEnd();
            i0.ɵɵelementEnd();
        }
    }
    // TODO ngMAterial change the way to work with color (.inform => background-color: primary; )
    var NotificationHandlerComponent = /** @class */ (function () {
        function NotificationHandlerComponent(snackBarRef, data) {
            this.snackBarRef = snackBarRef;
            this.data = data;
        }
        return NotificationHandlerComponent;
    }());
    NotificationHandlerComponent.ɵfac = function NotificationHandlerComponent_Factory(t) { return new (t || NotificationHandlerComponent)(i0.ɵɵdirectiveInject(i1.MatSnackBarRef), i0.ɵɵdirectiveInject(i1.MAT_SNACK_BAR_DATA)); };
    NotificationHandlerComponent.ɵcmp = i0.ɵɵdefineComponent({ type: NotificationHandlerComponent, selectors: [["ngy-notification-handler"]], decls: 4, vars: 2, consts: [[1, "container"], [2, "margin-left", "8px", "margin-right", "8px"], ["class", "dismiss", 4, "ngIf"], [1, "dismiss"], ["mat-mini-fab", "", 3, "click"]], template: function NotificationHandlerComponent_Template(rf, ctx) {
            if (rf & 1) {
                i0.ɵɵelementStart(0, "div", 0);
                i0.ɵɵelementStart(1, "div", 1);
                i0.ɵɵtext(2);
                i0.ɵɵelementEnd();
                i0.ɵɵtemplate(3, NotificationHandlerComponent_div_3_Template, 4, 0, "div", 2);
                i0.ɵɵelementEnd();
            }
            if (rf & 2) {
                i0.ɵɵadvance(2);
                i0.ɵɵtextInterpolate1(" ", ctx.data.message, " ");
                i0.ɵɵadvance(1);
                i0.ɵɵproperty("ngIf", ctx.data.allowDismiss);
            }
        }, directives: [i2.NgIf, i3.MatButton, i4.MatIcon], styles: [".container[_ngcontent-%COMP%]{align-items:center;display:flex;flex-direction:row}.dismiss[_ngcontent-%COMP%]{margin-left:auto}"] });
    /*@__PURE__*/ (function () {
        i0.ɵsetClassMetadata(NotificationHandlerComponent, [{
                type: i0.Component,
                args: [{
                        selector: 'ngy-notification-handler',
                        templateUrl: './notification-handler.component.html',
                        styleUrls: ['./notification-handler.component.scss'],
                    }]
            }], function () {
            return [{ type: i1.MatSnackBarRef }, { type: undefined, decorators: [{
                            type: i0.Inject,
                            args: [i1.MAT_SNACK_BAR_DATA]
                        }] }];
        }, null);
    })();

    (function (NotificationStyle) {
        NotificationStyle["default"] = "default-notification-overlay";
        NotificationStyle["error"] = "error-notification-overlay";
        NotificationStyle["success"] = "success-notification-overlay";
        NotificationStyle["info"] = "info-notification-overlay";
        NotificationStyle["warn"] = "warning-notification-overlay";
    })(exports.NotificationStyle || (exports.NotificationStyle = {}));

    var dismissibleNotificationStyle = [
        exports.NotificationStyle.error,
        exports.NotificationStyle.warn,
    ];
    var NotificationService = /** @class */ (function () {
        function NotificationService(snackBar) {
            this.snackBar = snackBar;
        }
        NotificationService.prototype.notify = function (notificationInfo) {
            var data = this.setDateWithDefault(notificationInfo);
            var options = this.setOptionsWithDefault(notificationInfo);
            this.snackBar.openFromComponent(NotificationHandlerComponent, Object.assign({ data: data }, options));
        };
        NotificationService.prototype.setDateWithDefault = function (notificationInfo) {
            return {
                message: notificationInfo.message,
                allowDismiss: notificationInfo.allowDismiss || dismissibleNotificationStyle.toString().includes(notificationInfo.kind),
            };
        };
        NotificationService.prototype.setOptionsWithDefault = function (notificationInfo) {
            var _a;
            var options = Object.assign({}, Object.assign({ duration: 2500 }, notificationInfo.options));
            options.panelClass = notificationInfo.kind
                ? [notificationInfo.kind]
                : [exports.NotificationStyle.default];
            options.panelClass = ((_a = notificationInfo.options) === null || _a === void 0 ? void 0 : _a.panelClass) ? __spread(options.panelClass, notificationInfo.options.panelClass) : options.panelClass;
            return options;
        };
        return NotificationService;
    }());
    NotificationService.ɵfac = function NotificationService_Factory(t) { return new (t || NotificationService)(i0.ɵɵinject(i1$1.MatSnackBar)); };
    NotificationService.ɵprov = i0.ɵɵdefineInjectable({ token: NotificationService, factory: NotificationService.ɵfac });
    /*@__PURE__*/ (function () {
        i0.ɵsetClassMetadata(NotificationService, [{
                type: i0.Injectable
            }], function () { return [{ type: i1$1.MatSnackBar }]; }, null);
    })();

    var NotificationHandlerFacadeService = /** @class */ (function () {
        function NotificationHandlerFacadeService(loggingService, notificationService) {
            this.loggingService = loggingService;
            this.notificationService = notificationService;
        }
        /**
         * LoggingService
         */
        NotificationHandlerFacadeService.prototype.logInformation = function (message) {
            return this.loggingService.logInformation(message);
        };
        NotificationHandlerFacadeService.prototype.logError = function (message) {
            return this.loggingService.logError(message);
        };
        NotificationHandlerFacadeService.prototype.logWarning = function (message) {
            return this.loggingService.logWarning(message);
        };
        NotificationHandlerFacadeService.prototype.logIt = function (message) {
            return this.loggingService.logIt(message);
        };
        /**
         * NotificationService
         */
        NotificationHandlerFacadeService.prototype.notify = function (notificationInfo) {
            return this.notificationService.notify(notificationInfo);
        };
        return NotificationHandlerFacadeService;
    }());
    NotificationHandlerFacadeService.ɵfac = function NotificationHandlerFacadeService_Factory(t) { return new (t || NotificationHandlerFacadeService)(i0.ɵɵinject(LoggingService), i0.ɵɵinject(NotificationService)); };
    NotificationHandlerFacadeService.ɵprov = i0.ɵɵdefineInjectable({ token: NotificationHandlerFacadeService, factory: NotificationHandlerFacadeService.ɵfac });
    /*@__PURE__*/ (function () {
        i0.ɵsetClassMetadata(NotificationHandlerFacadeService, [{
                type: i0.Injectable
            }], function () { return [{ type: LoggingService }, { type: NotificationService }]; }, null);
    })();

    var NotificationHandlerModule = /** @class */ (function () {
        function NotificationHandlerModule() {
        }
        return NotificationHandlerModule;
    }());
    NotificationHandlerModule.ɵmod = i0.ɵɵdefineNgModule({ type: NotificationHandlerModule });
    NotificationHandlerModule.ɵinj = i0.ɵɵdefineInjector({ factory: function NotificationHandlerModule_Factory(t) { return new (t || NotificationHandlerModule)(); }, providers: [
            LoggingService,
            NotificationService,
            NotificationHandlerFacadeService
        ], imports: [[
                i2.CommonModule,
                i1.MatSnackBarModule,
                i3.MatButtonModule,
                i4.MatIconModule,
            ]] });
    (function () {
        (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(NotificationHandlerModule, { declarations: [NotificationHandlerComponent], imports: [i2.CommonModule,
                i1.MatSnackBarModule,
                i3.MatButtonModule,
                i4.MatIconModule] });
    })();
    /*@__PURE__*/ (function () {
        i0.ɵsetClassMetadata(NotificationHandlerModule, [{
                type: i0.NgModule,
                args: [{
                        declarations: [
                            NotificationHandlerComponent
                        ],
                        imports: [
                            i2.CommonModule,
                            i1.MatSnackBarModule,
                            i3.MatButtonModule,
                            i4.MatIconModule,
                        ],
                        providers: [
                            LoggingService,
                            NotificationService,
                            NotificationHandlerFacadeService
                        ],
                        exports: [],
                        entryComponents: [NotificationHandlerComponent]
                    }]
            }], null, null);
    })();

    /*
     * Public API Surface of notification-handler
     */

    /**
     * Generated bundle index. Do not edit.
     */

    exports.NotificationHandlerFacadeService = NotificationHandlerFacadeService;
    exports.NotificationHandlerModule = NotificationHandlerModule;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
