// notification-handler/esm2015/lib/feature-log/logging.service.js
import { Injectable } from '@angular/core';
import * as i0 from "@angular/core";
// TODO: Log error to the server like Rollbar or Crashlytics
// TODO: treat the stack when error occurs
// TODO: set the application name
export class LoggingService {
    logError(message) {
        console.error(this.formatMessage(message));
    }
    logWarning(message) {
        console.warn(this.formatMessage(message));
    }
    logInformation(message) {
        console.log(`Info: ${this.formatMessage(message)}`);
    }
    logIt(message) {
        console.log(this.formatMessage(message));
    }
    formatMessage(message) {
        const now = new Date();
        return `${now.toLocaleString()}: LoggingService: ${message}`;
    }
}
LoggingService.ɵfac = function LoggingService_Factory(t) { return new (t || LoggingService)(); };
LoggingService.ɵprov = i0.ɵɵdefineInjectable({ token: LoggingService, factory: LoggingService.ɵfac });
/*@__PURE__*/ (function () { i0.ɵsetClassMetadata(LoggingService, [{
        type: Injectable
    }], null, null); })();
