// notification-handler/esm2015/lib/global-error-handler.js
import { Injectable } from '@angular/core';
import { HttpErrorResponse } from '@angular/common/http';
import { LoggingService } from './feature-log/logging.service';
import { NotificationService } from './feature-notify/notification.service';
import * as i0 from "@angular/core";
export class GlobalErrorHandler {
    constructor(injector) {
        this.injector = injector;
    }
    handleError(error) {
        const logger = this.injector.get(LoggingService);
        const notifier = this.injector.get(NotificationService);
        let errorDesc;
        if (error instanceof HttpErrorResponse) { // Server error
            errorDesc = this.getServerError(error);
        }
        else { // Client Error
            errorDesc = this.getClientError(error);
        }
        notifier.notify({ message: errorDesc.message });
        // Always log errors
        logger.logError(errorDesc.message); //errorDesc.stackTrace);
    }
    getServerError(error) {
        if (navigator.onLine) {
            return { message: error.message, stackTrace: error.error };
        }
        else {
            return { message: 'No Internet Connection', stackTrace: '' };
        }
    }
    getClientError(error) {
        if (error.message) {
            return { message: error.message, stackTrace: error.stack };
        }
        else {
            return { message: error.toString(), stackTrace: '' };
        }
    }
}
GlobalErrorHandler.ɵfac = function GlobalErrorHandler_Factory(t) { return new (t || GlobalErrorHandler)(i0.ɵɵinject(i0.Injector)); };
GlobalErrorHandler.ɵprov = i0.ɵɵdefineInjectable({ token: GlobalErrorHandler, factory: GlobalErrorHandler.ɵfac });
/*@__PURE__*/ (function () { i0.ɵsetClassMetadata(GlobalErrorHandler, [{
        type: Injectable
    }], function () { return [{ type: i0.Injector }]; }, null); })();
