// notification-handler/esm2015/lib/notification-handler-facade.service.js
import { Injectable } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "./feature-log";
import * as i2 from "./feature-notify";
export class NotificationHandlerFacadeService {
    constructor(loggingService, notificationService) {
        this.loggingService = loggingService;
        this.notificationService = notificationService;
    }
    /**
     * LoggingService
     */
    logInformation(message) {
        return this.loggingService.logInformation(message);
    }
    logError(message) {
        return this.loggingService.logError(message);
    }
    logWarning(message) {
        return this.loggingService.logWarning(message);
    }
    logIt(message) {
        return this.loggingService.logIt(message);
    }
    /**
     * NotificationService
     */
    notify(notificationInfo) {
        return this.notificationService.notify(notificationInfo);
    }
}
NotificationHandlerFacadeService.ɵfac = function NotificationHandlerFacadeService_Factory(t) { return new (t || NotificationHandlerFacadeService)(i0.ɵɵinject(i1.LoggingService), i0.ɵɵinject(i2.NotificationService)); };
NotificationHandlerFacadeService.ɵprov = i0.ɵɵdefineInjectable({ token: NotificationHandlerFacadeService, factory: NotificationHandlerFacadeService.ɵfac });
/*@__PURE__*/ (function () { i0.ɵsetClassMetadata(NotificationHandlerFacadeService, [{
        type: Injectable
    }], function () { return [{ type: i1.LoggingService }, { type: i2.NotificationService }]; }, null); })();
