// notification-handler/esm2015/lib/notification-handler.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { MatSnackBarModule } from '@angular/material/snack-bar';
import { MatIconModule } from '@angular/material/icon';
import { NotificationHandlerComponent } from './feature-notify/notification-handler/notification-handler.component';
import { NotificationHandlerFacadeService } from './notification-handler-facade.service';
import { LoggingService } from './feature-log';
import { NotificationService } from './feature-notify';
import { MatButtonModule } from '@angular/material/button';
import * as i0 from "@angular/core";
export class NotificationHandlerModule {
}
NotificationHandlerModule.ɵmod = i0.ɵɵdefineNgModule({ type: NotificationHandlerModule });
NotificationHandlerModule.ɵinj = i0.ɵɵdefineInjector({ factory: function NotificationHandlerModule_Factory(t) { return new (t || NotificationHandlerModule)(); }, providers: [
        LoggingService,
        NotificationService,
        NotificationHandlerFacadeService
    ], imports: [[
            CommonModule,
            MatSnackBarModule,
            MatButtonModule,
            MatIconModule,
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(NotificationHandlerModule, { declarations: [NotificationHandlerComponent], imports: [CommonModule,
        MatSnackBarModule,
        MatButtonModule,
        MatIconModule] }); })();
/*@__PURE__*/ (function () { i0.ɵsetClassMetadata(NotificationHandlerModule, [{
        type: NgModule,
        args: [{
                declarations: [
                    NotificationHandlerComponent
                ],
                imports: [
                    CommonModule,
                    MatSnackBarModule,
                    MatButtonModule,
                    MatIconModule,
                ],
                providers: [
                    LoggingService,
                    NotificationService,
                    NotificationHandlerFacadeService
                ],
                exports: [],
                entryComponents: [NotificationHandlerComponent]
            }]
    }], null, null); })();
