// notification-handler/esm2015/lib/feature-notify/notification-handler/notification-handler.component.js
import { Component, Inject } from '@angular/core';
import { MAT_SNACK_BAR_DATA, } from '@angular/material/snack-bar';
import * as i0 from "@angular/core";
import * as i1 from "@angular/material/snack-bar";
import * as i2 from "@angular/common";
import * as i3 from "@angular/material/button";
import * as i4 from "@angular/material/icon";
function NotificationHandlerComponent_div_3_Template(rf, ctx) { if (rf & 1) {
    const _r2 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "div", 3);
    i0.ɵɵelementStart(1, "button", 4);
    i0.ɵɵlistener("click", function NotificationHandlerComponent_div_3_Template_button_click_1_listener() { i0.ɵɵrestoreView(_r2); const ctx_r1 = i0.ɵɵnextContext(); return ctx_r1.snackBarRef.dismiss(); });
    i0.ɵɵelementStart(2, "mat-icon");
    i0.ɵɵtext(3, "close");
    i0.ɵɵelementEnd();
    i0.ɵɵelementEnd();
    i0.ɵɵelementEnd();
} }
// TODO ngMAterial change the way to work with color (.inform => background-color: primary; )
export class NotificationHandlerComponent {
    constructor(snackBarRef, data) {
        this.snackBarRef = snackBarRef;
        this.data = data;
    }
}
NotificationHandlerComponent.ɵfac = function NotificationHandlerComponent_Factory(t) { return new (t || NotificationHandlerComponent)(i0.ɵɵdirectiveInject(i1.MatSnackBarRef), i0.ɵɵdirectiveInject(MAT_SNACK_BAR_DATA)); };
NotificationHandlerComponent.ɵcmp = i0.ɵɵdefineComponent({ type: NotificationHandlerComponent, selectors: [["ngy-notification-handler"]], decls: 4, vars: 2, consts: [[1, "container"], [2, "margin-left", "8px", "margin-right", "8px"], ["class", "dismiss", 4, "ngIf"], [1, "dismiss"], ["mat-mini-fab", "", 3, "click"]], template: function NotificationHandlerComponent_Template(rf, ctx) { if (rf & 1) {
        i0.ɵɵelementStart(0, "div", 0);
        i0.ɵɵelementStart(1, "div", 1);
        i0.ɵɵtext(2);
        i0.ɵɵelementEnd();
        i0.ɵɵtemplate(3, NotificationHandlerComponent_div_3_Template, 4, 0, "div", 2);
        i0.ɵɵelementEnd();
    } if (rf & 2) {
        i0.ɵɵadvance(2);
        i0.ɵɵtextInterpolate1(" ", ctx.data.message, " ");
        i0.ɵɵadvance(1);
        i0.ɵɵproperty("ngIf", ctx.data.allowDismiss);
    } }, directives: [i2.NgIf, i3.MatButton, i4.MatIcon], styles: [".container[_ngcontent-%COMP%]{align-items:center;display:flex;flex-direction:row}.dismiss[_ngcontent-%COMP%]{margin-left:auto}"] });
/*@__PURE__*/ (function () { i0.ɵsetClassMetadata(NotificationHandlerComponent, [{
        type: Component,
        args: [{
                selector: 'ngy-notification-handler',
                templateUrl: './notification-handler.component.html',
                styleUrls: ['./notification-handler.component.scss'],
            }]
    }], function () { return [{ type: i1.MatSnackBarRef }, { type: undefined, decorators: [{
                type: Inject,
                args: [MAT_SNACK_BAR_DATA]
            }] }]; }, null); })();
