// notification-handler/esm2015/lib/feature-notify/notification.service.js
import { Injectable } from '@angular/core';
import { NotificationHandlerComponent } from './notification-handler/notification-handler.component';
import { NotificationStyle } from './notifycation-style.enum';
import * as i0 from "@angular/core";
import * as i1 from "@angular/material//snack-bar";
const dismissibleNotificationStyle = [
    NotificationStyle.error,
    NotificationStyle.warn,
];
export class NotificationService {
    constructor(snackBar) {
        this.snackBar = snackBar;
    }
    notify(notificationInfo) {
        const data = this.setDateWithDefault(notificationInfo);
        const options = this.setOptionsWithDefault(notificationInfo);
        this.snackBar.openFromComponent(NotificationHandlerComponent, Object.assign({ data }, options));
    }
    setDateWithDefault(notificationInfo) {
        return {
            message: notificationInfo.message,
            allowDismiss: notificationInfo.allowDismiss || dismissibleNotificationStyle.toString().includes(notificationInfo.kind),
        };
    }
    setOptionsWithDefault(notificationInfo) {
        var _a;
        const options = Object.assign({}, Object.assign({ duration: 2500 }, notificationInfo.options));
        options.panelClass = notificationInfo.kind
            ? [notificationInfo.kind]
            : [NotificationStyle.default];
        options.panelClass = ((_a = notificationInfo.options) === null || _a === void 0 ? void 0 : _a.panelClass) ? [...options.panelClass, ...notificationInfo.options.panelClass]
            : options.panelClass;
        return options;
    }
}
NotificationService.ɵfac = function NotificationService_Factory(t) { return new (t || NotificationService)(i0.ɵɵinject(i1.MatSnackBar)); };
NotificationService.ɵprov = i0.ɵɵdefineInjectable({ token: NotificationService, factory: NotificationService.ɵfac });
/*@__PURE__*/ (function () { i0.ɵsetClassMetadata(NotificationService, [{
        type: Injectable
    }], function () { return [{ type: i1.MatSnackBar }]; }, null); })();
